### PR TITLE
Make some entity events cancellable

### PR DESCRIPTION
**Pull request by LoboMetalurgico**

### CODE REVIEW

File: src/main/java/cn/nukkit/entity/Entity.java [hunk 1]
```diff
@@ -596,15 +596,26 @@ protected final void init(FullChunk chunk, CompoundTag nbt) {
...
+
+            this.server.getPluginManager().callEvent(event);
+
+            if (event.isCancelled()) {
+                this.close(false);
+            } else {
+                this.scheduleUpdate();
+            }
+        } catch(Exception e) {
+            this.close(false);
```
Comment (Gabriel8579) [inline src/main/java/cn/nukkit/entity/Entity.java:617]: Could you add some console log into this point?

**Review by Gabriel8579 — CHANGES_REQUESTED** [review]

File: src/main/java/cn/nukkit/entity/Entity.java (hunk 1 above)
Comment (joserobjr) [reply src/main/java/cn/nukkit/entity/Entity.java:617]: I think it's better to re-throw the exception instead

Comment (joserobjr) [reply src/main/java/cn/nukkit/entity/Entity.java:617]: that would let the caller handle the exception

Comment (LoboMetalurgico) [reply src/main/java/cn/nukkit/entity/Entity.java:617]: Yes, thank you for reminding me of this.

Comment (Gabriel8579) [reply src/main/java/cn/nukkit/entity/Entity.java:617]: I agree

File: src/main/java/cn/nukkit/entity/Entity.java [hunk 2]
```diff
@@ -596,15 +596,27 @@ protected final void init(FullChunk chunk, CompoundTag nbt) {
...
+            this.server.getPluginManager().callEvent(event);
+
+            if (event.isCancelled()) {
+                this.close(false);
+            } else {
+                this.scheduleUpdate();
+            }
+        } catch(Exception e) {
+            throw new Exception(e);
+            this.close(false);
```
Comment (Gabriel8579) [inline src/main/java/cn/nukkit/entity/Entity.java:618]:
```suggestion
            this.close(false);
            throw e;
            
```

**Review by Gabriel8579 — CHANGES_REQUESTED** [review]

**Review by Gabriel8579 — APPROVED** [review]
Looks good now, thank you!

**Pull request merged**